sympy/series/sequences.py

```diff
@@ -0,0 +1,522 @@
+
+    @property
+    def interval(self):
+        """The interval in which the sequence is defined"""
+        return self._interval
+
+    @property
+    def _interval(self):
+        raise NotImplementedError("(%s)._interval" % self)
+
+    @property
+    def start(self):
+        """The starting point of the sequence. This point is included"""
+        return self._start
+
+    @property
+    def _start(self):
+        raise NotImplementedError("(%s)._start" % self)
+
+    @property
+    def stop(self):
+        """The ending point of the sequence. This point is included"""
+        return self._stop
+
+    @property
+    def _stop(self):
+        raise NotImplementedError("(%s)._stop" % self)
+
+    @property
+    def step(self):
+        """Increase points by step"""
+        return self._step
+
+    @property
+    def _step(self):
+        raise NotImplementedError("(%s)._step" % self)
+
+    @property
+    def length(self):
+        """Length of the sequence"""
+        return self._length
+
+    @property
+    def _length(self):
+        raise NotImplementedError("(%s)._length" % self)
+
+    @property
+    def dummy(self):
+        """Returns a tuple of variables that are bounded"""
+        return self._dummy
+
+    @property
+    def _dummy(self):
+        return ()
+
+    def coeff(self, i):
+        """Returns the coefficient at point i"""
+        if i < self.start or i > self.stop:
+            raise IndexError("Index %s out of bounds %s" %(i, self.interval))
+        return self._eval_coeff(i)
+
+    def _eval_coeff(self, i):
+        raise NotImplementedError(filldedent(""" The _eval_coeff method should\
+                                             be added to %s to return\
+                                             coefficient so it is available\
+                                             when coeff calls it."""\
+                                             % self.func))
+
+    def _ith_point(self, i, step=None):
+        """
+        Returns the i'th point of a sequence
+        If start point is negative infinity, point is returned from the end.
+        Assumes the first point to be indexed zero.
+
+        Examples
+        =========
+
+        >>> from sympy import oo
+        >>> from sympy.series.sequences import SeqExpr
+
+        bounded
+
+        >>> SeqExpr((1, 2, 3), (-10, 10))._ith_point(0)
+        -10
+        >>> SeqExpr((1, 2, 3), (-10, 10))._ith_point(5)
+        -5
+        >>> SeqExpr((1, 2, 3), (-10, 10, 2))._ith_point(5)
+        0
+
+        End is at infinity
+
+        >>> SeqExpr((1, 2, 3), (0, oo))._ith_point(5)
+        5
+        >>> SeqExpr((1, 2, 3), (0, oo, 2))._ith_point(5)
+        10
+
+        Starts at negative infinity
+
+        >>> SeqExpr((1, 2, 3), (-oo, 0))._ith_point(5)
+        -5
+        >>> SeqExpr((1, 2, 3), (-oo, 0, 2))._ith_point(5)
+        -10
+        """
+        if self.start is S.NegativeInfinity:
+            initial = self.stop
```

Comment:
The reason I disallowed both end-points to be unbounded is due to ``__getitem__``. Now if a sequence is defined on ``(-oo, oo)``, what will 0th index be? I have tried to define 0th index as the starting point of the sequence. This is for convenience, say a sequence is defined in ``(5, 10)`` then 0th index will be value at point 5.